src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -686,34 +878,196 @@ void fmi3_xml_variable_free_internals(jm_callbacks* callbacks, fmi3_xml_variable
     }
 }
 
-int fmi3_xml_handle_Variable(fmi3_xml_parser_context_t *context, const char* data) {
```

Comment:
I moved out some long functions for getting variability etc. attributes, and that makes the diff strange. I suggest to look at the actual code before trying to read this diff.